HOW-TO-READ-ERRORS.md

```diff
@@ -12,12 +12,12 @@ error[E0277]: the trait bound ...
    |
    ...
 ```
-It means that provider cannot resolve that `SERVICE`. Possible reasons are:
+It means that the provider cannot resolve that `SERVICE`. Possible reasons are:
 1. You do not register this dependency in `ServiceProvider`.
-2. If you register the `Service`, than check by which lifetime:
-    - If it registered using `Transient` lifetime (`add_transient` method), than it can be resolved only by ownership,
-    so check that it not resolved by reference.
-    - If it registered using `Singleton` lifetime (`add_singleton` method), than it can be resolved by clone in cases when
-    `Service` implement `DependencyClone`, so if you want to resolve it by clone, check that it implement `DependencyClone`.
-    It implement by default for `Rc`, `Arc` and `&T`. Otherwise it can be resolved only by reference.
+2. If you register the `Service`, then check by which lifetime:
+    - If it is registered using `Transient` lifetime (`add_transient` method), then it can be resolved only by ownership,
+    so check that it is not resolved by reference.
+    - If it is registered using `Singleton` lifetime (`add_singleton` method), then it can be resolved by clone in cases when
+    `Service` implement `DependencyClone`, so if you want to resolve it by clone, check that it implements `DependencyClone`.
+    It is implemented by default for `Rc`, `Arc` and `&T`. Otherwise it can be resolved only by reference.
```

Comment:
Similar to the above list item, the wording of this could be improved. What do you think of this?
```
2. `SERVICE` was registered as a dependency in `ServiceProvider`, but the resolved lifetime is wrong:
  a. If it is registered as a `Transient` dependency (e.g. registered by `add_transient()`), then it can only be resolved by ownership and not by reference.
  b. If it is registered as a `Singleton` dependency (e.g. registered by 'add_singleton()`), then it can be resolved by ownership when `SERVICE` implements `DependencyClone`, otherwise it is only resolvable by reference. `Rc`, `Arc` and `&T` implement `DependencyClone` by default.
```